smithy-aws-apigateway-traits/src/main/java/software/amazon/smithy/aws/apigateway/traits/IntegrationTrait.java

```diff
@@ -336,12 +337,14 @@ protected ObjectNode createNode() {
         NodeMapper mapper = new NodeMapper();
         mapper.disableToNodeForClass(IntegrationTrait.class);
         mapper.setOmitEmptyValues(true);
-        return mapper.serialize(this).expectObjectNode();
+        ObjectNode.Builder builder = mapper.serialize(this).expectObjectNode().toBuilder();
+        return builder.sourceLocation(getSourceLocation()).build();
```

Comment:
@JordonPhillips this still is necessary after https://github.com/awslabs/smithy/pull/865 right?

Reply:
Nope

Reply:
I think it is. Because of https://github.com/awslabs/smithy/blob/main/smithy-model/src/main/java/software/amazon/smithy/model/node/NodeMapper.java#L315. You refer to it here https://github.com/awslabs/smithy/pull/865#discussion_r672154242.

Reply:
It isn't. This implementation of `ToNode` is using the `NodeMapper`, which will set it for you. The implementation is identical to the same method in [this test trait](https://github.com/awslabs/smithy/blob/main/smithy-model/src/test/java/software/amazon/smithy/model/node/NodeMapperTest.java#L1282-L1287). I modeled that test class after this exact trait

Reply:
You are right. I did not notice the `disableToNodeForClass` part. I was also confused by ObjectNode.toBuilder() not propagating the sourceLocation, which affected my local testing.